src/components/SwapWormholeDialog.js

```diff
@@ -0,0 +1,825 @@
+  // by trading on swap market.
+  async function convert() {
+    const swapAmount = new BN(parsedAmount * 10 ** balanceInfo.decimals);
+    // 1 for 1 swap, subtracting out taker fee.
+    const minExpectedAmount = swapAmount.mul(new BN(9978)).div(new BN(10000));
+
+    const [vaultSigner] = await getVaultOwnerAndNonce(
+      market.account._decoded.ownAddress,
+    );
+    let [openOrders, needsCreateOpenOrders] = await (async () => {
+      let openOrders = await OpenOrders.findForOwner(
+        swapClient.provider.connection,
+        wallet.publicKey,
+        DEX_PROGRAM_ID,
+      );
+
+      // If we have an open orderes account use it. It doesn't matter which
+      // one we use.
+      const addr = openOrders[0] ? openOrders[0].address : undefined;
+      return [addr, addr !== undefined];
+    })();
+    let signers = [];
+
+    // Build the transaction.
+    const tx = new Transaction();
+
+    // Create the wormhole associated token account, if needed.
+    let _wormholeTokenAddr = wormholeTokenAddr;
+    if (!_wormholeTokenAddr) {
+      const [ix, addr] = await createAssociatedTokenAccountIx(
+        wallet.publicKey,
+        wallet.publicKey,
+        wormholeMintAddr,
+      );
+      tx.add(ix);
+      _wormholeTokenAddr = addr;
+    } else {
+      _wormholeTokenAddr = new PublicKey(_wormholeTokenAddr.toString());
+    }
+
+    // Create the open orders account, if needed.
+    if (needsCreateOpenOrders) {
+      const _openOrders = new Account();
+      openOrders = _openOrders.publicKey;
+      signers.push(_openOrders);
+      tx.add(
+        await OpenOrders.makeCreateAccountTransaction(
+          swapClient.provider.connection,
+          market.account._decoded.ownAddress,
+          swapClient.provider.wallet.publicKey,
+          openOrders,
+          DEX_PROGRAM_ID,
+        ),
+      );
+    }
+    // Execute the swap.
+    tx.add(
+      swapClient.instruction.swap(Side.Ask, swapAmount, minExpectedAmount, {
+        accounts: {
+          market: {
+            market: market.account._decoded.ownAddress,
+            requestQueue: market.account._decoded.requestQueue,
+            eventQueue: market.account._decoded.eventQueue,
+            bids: market.account._decoded.bids,
+            asks: market.account._decoded.asks,
+            coinVault: market.account._decoded.baseVault,
+            pcVault: market.account._decoded.quoteVault,
+            vaultSigner,
+            openOrders,
+            orderPayerTokenAccount: publicKey,
+            coinWallet: publicKey,
+          },
+          pcWallet: _wormholeTokenAddr,
+          authority: swapClient.provider.wallet.publicKey,
+          dexProgram: DEX_PROGRAM_ID,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+      }),
+    );
+
+    // Close the open orders account, if needed.
+    if (needsCreateOpenOrders) {
+      // TODO: enable once the dex supports this.
```

Comment:
Open PR for this against the dex here https://github.com/project-serum/serum-dex/pull/112. Not a blocker but would be nice to have.